sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -334,6 +334,19 @@ abstract class RapidsBufferStore(
       }
     }
 
+    override def copyToMemoryBuffer(srcOffset: Long, dst: MemoryBuffer, dstOffset: Long,
+        length: Long, stream: Cuda.Stream): Unit = {
+      withResource(getMemoryBuffer) { memBuff =>
+        dst match {
+          case _: HostMemoryBuffer =>
+            dst.copyFromMemoryBuffer(dstOffset, memBuff, srcOffset, length, stream)
```

Comment:
Should this just use the `Async` version? Especially if `dst` is a pinned buffer, I believe that pinned copy can be done asynchronously.

Reply:
In the original code `BufferSendState` was doing a `CudaUtil.copy` instead of `copyAsync`, so I wanted to preserve the existing behavior. We can certainly change it here if you think it is appropriate.

Reply:
OK, maybe it can be a follow up to keep changes to a minimum. Agree with it not being part of this PR then.

Reply:
Can you add a TODO here please?

Reply:
Done.